restapi.go

```diff
@@ -2545,6 +2545,62 @@ func (s *Session) ApplicationCommands(appID, guildID string) (cmd []*Application
 	return
 }
 
+// GuildApplicationCommandsPermissions receives all permissions for application commands that have any permissions
+// appID       : The application ID
+// guildID     : Guild ID to retrieve all guild-specific permissions for application commands in the guild, both global and guild specific
+func (s *Session) GuildApplicationCommandsPermissions(appID, guildID string) (permissions []*GuildApplicationCommandPermissions) {
+	endpoint := EndpointApplicationCommandsGuildPermissions(appID, guildID)
+
+	body, err := s.RequestWithBucketID("GET", endpoint, nil, endpoint)
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &permissions)
+	return
+}
+
+// ApplicationCommandPermissions receives all permissions for application commands that have any permissions
+// appID       : The Application ID
+// guildID     : The guild ID containing the application command
+// cmdID       : The command ID to retrieve the permissions of
+func (s *Session) ApplicationCommandPermissions(appID, guildID, cmdID string) (permissions GuildApplicationCommandPermissions) {
+	endpoint := EndpointApplicationCommandPermissions(appID, guildID, cmdID)
+
+	body, err := s.RequestWithBucketID("GET", endpoint, nil, endpoint)
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &permissions)
+	return
+}
+
+// ApplicationCommandEditPermissions edits the permissions of an application command
+// appID       : The Application ID
+// guildID     : The guild ID containing the application command
+// cmdID       : The command ID to edit the permissions of
+// permissions : An object containing a list of permissions for the application command
+func (s *Session) ApplicationCommandEditPermissions(appID, guildID, cmdID string, permissions ApplicationCommandPermissionsList) (err error) {
+	endpoint := EndpointApplicationCommandPermissions(appID, guildID, cmdID)
+
+	_, err = s.RequestWithBucketID("PUT", endpoint, permissions, endpoint)
+
+	return
+}
+
+// ApplicationCommandBatchEditPermissions edits the permissions of a batch of commands
+// appID       : The Application ID
+// guildID     : The guild ID to batch edit commands of
+// permissions : A list of permissions paired with a command ID, guild ID, and application ID per application command
+func (s *Session) ApplicationCommandBatchEditPermissions(appID, guildID string, permissions []GuildApplicationCommandPermissions) (err error) {
```

Comment:
It seems like this library mostly puts the verb at the end of the method name, as in `ApplicationCommandBulkOverwrite`. This same comment applies to `ApplicationCommandEditPermissions` above.

Reply:
Thanks for your feedback, I've reflected it in the new change. I've kept the naming as BatchEdit as the API docs specifically mention it as a batch, not a bulk.